### ISSUE TITLE
Implement DownsampleFactorMax.R_op

### ISSUE THREAD

**Issue by nouiz** [Theano/Theano#305]
This was requested by a user in theano-dev mailing list in a thread with this title [[theano-dev] R-op DownsampleFactorMax](https://groups.google.com/d/topic/theano-dev/y6iwP0NhF9k/discussion)

**Comment by rodrigob** [comment]
I am currently blocked by this issue. Any chance this feature to be implemented ? (4 years old ticket...)

**Comment by nouiz** [comment]
this old PR https://github.com/Theano/Theano/pull/622/files have it mostly finished. I'm not sure why it wasn't finished, probably it got bumped by othe prio.

That PR included the rop of convolution that have been finished. Can you help us? Can you take this PR, remove the commit for the convolution rebase it on master and make a new PR? I'll push to have someone more knowledgeable about this to review it.

**Comment by rodrigob** [comment]
I will give it a shot....

**Comment by rodrigob** [comment]
Here is the shot. Clearly this will not flight by itself. It will need someone that is familiar with Theano internals.

**Comment by rodrigob** [comment]
> I'll push to have someone more knowledgeable about this to review it.
@nouiz who would that be ?

**Comment by rodrigob** [comment]
Looks like this issue is now being tackled via #4781

**Comment by nouiz** [comment]
done in https://github.com/Theano/Theano/pull/5323
